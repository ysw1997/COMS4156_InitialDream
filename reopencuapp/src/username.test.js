import React from 'react'
import Square from './username'
import { create } from 'react-test-renderer'

describe('Square Simple Snapshot Test', () => {
  test('Testing square', () => {
    const tree = create(<Square />)
    expect(tree.toJSON()).toMatchSnapshot()
  })
})

describe('Square className is affected by isWinningSquare prop', () => {
  test('props.isWinningSquare is false, className should be "square"', () => {
    const tree = create(<Square isWinningSquare={false} />)

    expect(tree.root.findByType('button').props.className).toEqual('square')
  }),
  test('props.isWinningSquare is true, className should be "square winning-square"', () => {
    const tree = create(<Square isWinningSquare />)

    expect(tree.root.findByType('button').props.className).toEqual('square winning-square')
  })
})

test('props.onClick is called when button is clicked', () => {
  const fn = jest.fn()
  const tree = create(<Square onClick={fn} />)
  // Simulate button click
  const button = tree.root.findByType('button')
  button.props.onClick()
  // Verify callback is invoked
  expect(fn.mock.calls.length).toBe(1)
})

describe('Square Simple Snapshot Test', () => {
  test('Testing square', () => {
    const tree = create(<Square />)
    expect(tree.toJSON()).toMatchSnapshot()
  })
})

describe('Square className is affected by isWinningSquare prop', () => {
  test('props.isWinningSquare is false, className should be "square"', () => {
    const tree = create(<Square isWinningSquare={false} />)

    expect(tree.root.findByType('button').props.className).toEqual('square')
  }),
  test('props.isWinningSquare is true, className should be "square winning-square"', () => {
    const tree = create(<Square isWinningSquare />)

    expect(tree.root.findByType('button').props.className).toEqual('square winning-square')
  })
})

test('props.onClick is called when button is clicked', () => {
  const fn = jest.fn()
  const tree = create(<Square onClick={fn} />)
  // Simulate button click
  const button = tree.root.findByType('button')
  button.props.onClick()
  // Verify callback is invoked
  expect(fn.mock.calls.length).toBe(1)
})
