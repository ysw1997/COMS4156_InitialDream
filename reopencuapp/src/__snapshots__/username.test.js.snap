// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square Simple Snapshot Test Testing square 1`] = `
<button
  className="square"
  onClick={[Function]}
/>
`;

exports[`Square Simple Snapshot Test Testing square 2`] = `
<button
  className="square"
  onClick={[Function]}
/>
`;
